oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/token/OAuth2AuthorizationGrantContext.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.server.authorization.token;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.springframework.security.oauth2.server.authorization.client.RegisteredClient;
+import org.springframework.util.Assert;
+
+/**
+ * Context describing current status of OAuth2 authorization. User by a {@link OAuth2TokenIssuer} to issue a new OAuth2 token.
+ *
+ * Different implementations of {@link OAuth2TokenIssuer} may decide to use different properties of
+ * the context.
+ *
+ * @author Alexey Nesterov
+ * @since 0.1.0
+ */
+public class OAuth2AuthorizationGrantContext {
+
+	private final String principalName;
+	private final RegisteredClient registeredClient;
+	private final Set<String> scopes;
```

Comment:
`scopes` is optional as per spec so we should change this field to be more generic, e.g. `claims`? During user consent, the user may authorize more than just `scopes`, they may authorize specific claims, e.g. `email`.